langchain/retrievers/sql_database.py

```diff
@@ -0,0 +1,42 @@
+"""SQL Database Chain Retriever"""
+from typing import Any, Dict, List
+
+from langchain.callbacks.manager import (
+    AsyncCallbackManagerForRetrieverRun,
+    CallbackManagerForRetrieverRun,
+)
+from langchain.chains.sql_database.base import SQLDatabaseChain
+from langchain.schema import BaseRetriever, Document
+
+
+class SQLDatabaseChainRetriever(BaseRetriever):
+    """Retriever that uses SQLDatabase as Retriever"""
+
+    sql_db_chain: SQLDatabaseChain
+    """SQL Database Chain"""
+    page_content_key: str = "content"
+    """column name for page content of documents"""
+
+    def _get_relevant_documents(
+        self,
+        query: str,
+        *,
+        run_manager: CallbackManagerForRetrieverRun,
+        **kwargs: Any,
+    ) -> List[Document]:
+        if not self.sql_db_chain.native_format:
```

Comment:
probably something that could be checked at instantiation

Reply:
Revised with validator to check at instantiation.